README.md

```diff
@@ -146,6 +146,140 @@ Module blocks could be transpiled to either data URLs, or to a module in a separ
+
+### Can you _statically_ import other modules?
+
+Yes. Just like with a separate file. This is completely valid:
+
+```js
+const m = module {
+  import myApiWrapper from "./api-wrapper.js";
+
+  await someTopLevelAwaitLogic();
+}
+```
+
+### What about a nice shorthand for modules that just contain one function?
+
+Something like this?
+
+```js
+const m = module (x) => x**2;
+
+// equivalent to
+
+const m = module {
+  export default (x) => x**2;
+}
+```
+
+It is unclear whether this will actually be a common pattern, as it not only requires the module to have exactly one function, but also that it does not need any static imports. For now we plan to leave the MVP without such a shorthand and observe what kind of usage patterns emerge. If import-free single-function modules become a frequently recurring pattern, a module shorthand seems like a good idea to add.
+
+### Can Module Blocks help with bundling?
+
+Mostly no. In _some_ scenarios Module Blocks can be sufficient to bundle:
+
+```js
+const count = module {
+  let i = 0;
+
+  export function count() {
+    i++;
+    return i;
+  }
+};
+
+const uppercase = module {
+  export function uppercase(string) {
+    return string.toUpperCase();
+  }
+};
+
+const { count } = await import(count);
+const { uppercase } = await import(uppercase);
+
+console.log(count()); // 1
+console.log(uppercase("daniel")); // "DANIEL"
+```
+
+In the _general_ case, however, modules need to refer to each other. For that to work Module Blocks would need to be able to close over modules, which they can’t do:
+
+```js
+const count = module {
+  let i = 0;
+
+  export function count() {
+    i++;
+    return i;
+  }
+};
+
+const uppercase = module {
+  export function uppercase(string) {
+    return string.toUpperCase();
+  }
+};
+
+const combined = module {
+  const { count } = await import(count);
+  const { uppercase } = await import(uppercase);
+
+  console.log(count()); // 1
+  console.log(uppercase("daniel")); // "DANIEL"
+};
+
+// ReferenceError as we can't close over count or uppercase!!
+```
+
+To address the bundling problem, Dan Ehrenberg is maintaining a separate [proposal/idea][js module bundles].
+
+### What about Blöcks?
+
+[Blöcks] has been archived. Module blocks is probably a better fit for JavaScript for a bunch of reasons:
+- Blöcks was trying to introduce a new type of function. Both imply that you can close over/capture values outside that scope. We tried to allow that in Blöcks (because it is expected) which turned out to be a can of worms.
+- Instead, Modules are well-explored, well-specified and well-understood by tooling, engines and developers. A lot of questions we had to worry about in Blöcks are naturally resolved through prior work in the modules space (e.g a module can only reference the global scope and do imports).
+- Modules already have a caching mechanism.
+- Modules are easier to  backwards compatible by making module blocks ObjectURL-able.
+
+### What about TypeScript?
```

Comment:
Hmm, I think the answer here should actually be going in the direction of what @rricard has proposed: the syntax could be something like `module<GlobalInterface> { }`. We should talk about how our plan is to work with TypeScript to find a good solution here. It's possible that they could read this as brushing off the concern.

Reply:
Suggested answer:

> We've heard concerns from the TypeScript team that it could be difficult to type access to the global object within a module blocks. Unfortunately, this is part of a bigger pattern with TypeScript:
>
> It is notoriously difficult to define what kind of scope a TypeScript file should be executed in (Main thread vs worker vs service worker), which is often solved by having multiple tsconfig.json file and composited projects. In that scenario, it’s even harder to have code that is shared across these TS projects.
> 
> When communicating with a Worker, you already need to force a type on the `event.data` to bring typing to the communication channel.
>
> All in all, it's hard to judge how much worse or more complicated Module Blocks makes the typing situation.
>
> Nevertheless, we're thinking about this problem and in early discussions with the TypeScript team about possible solutions, such as a TS syntax for annotating the type of the global object for a module block, such as `module<GlobalInterface> { }`